glslc/src/file_compiler.cc

```diff
@@ -96,8 +99,24 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
     return false;
   }
 
+  // Cache the compilation output, its content might be overwritten by
```

Comment:
This always makes a copy of the compilation results, even in the case there is no need.

you could avoid the copy as follows.
compilation_output could be a stringpiece that, for now points to the result.GetData(), result.GetLength.
The return value of DumpDependencyInfo could be a std::pair<bool, std::string>,
where the first part is the current return value (pass/fail), and the second part is the dependency output string if it's generated as compilation output, and an empty string otherwise.
Then if that second parameter is non-empty, then set the string-piece to point to that string.  (But make sure the storage stays around long enough to e used when dumping the output. That is, declare the std::pair result variable before checking if the dependency info handler exists.)

That way you never make a copy of the compilation output when you don't need to.